feat: create basic layout

rina214 opened the pull request:
- 헤더와 페이지, 하단 내비게이션을 만들었습니다.

rina214 committed:
feat: create basic layout

- 헤더와 페이지, 하단 내비게이션을 만들었습니다.

Johnie-Yeo commented on web/src/App.js:
```diff
@@ -1,26 +1,60 @@
+  }
+  render(){
+    var page = <HomePage></HomePage>;
+    if(this.state.mode === 1){
```
magic number에 대해서 한번 봐 주시면 좋을 것 같습니다.

Johnie-Yeo commented on web/src/App.js:
```diff
@@ -1,26 +1,60 @@
+
+    return (
+      <div className="App">
+        <Header className="App-header"
```
여기서 Header와 BottomMenu를 별도의 Layout이라는 컴포넌트로 분리한다면 훨씬 더 깔끔하게 관리 될 수 있을 것 같아요!
React의 children이라는 props와 함께 사용해 보시는 걸 추천드립니다.

Johnie-Yeo commented on web/src/App.js:
```diff
@@ -1,26 +1,60 @@
+class App extends Component {
+  constructor(props){ //render보다 먼저 실행됨, 초기화 담당
+    super(props);
+    this.state = { //내부적으로 사용할 상태 -> state
```
변화하지 않는 단순한 데이터 집합일 것 같은데 굳이 state를 사용할 이유가 있을까요?

Johnie-Yeo commented on web/src/App.js:
```diff
@@ -1,26 +1,60 @@
+    }
+  }
+  render(){
+    var page = <HomePage></HomePage>;
```
React의 기본 철학을 고려하면 var의 사용은 맞지 않습니다.
immutable한 데이터 관리를 위해서 const로 관리해 주세요.

Johnie-Yeo commented on web/src/components/BottomMenu/index.js:
```diff
@@ -0,0 +1,29 @@
+        var lists = []
+        var data = this.props.data;
+        var i = 0;
+        while(i < data.length) {
```
ES6의 Array.map을 활용하면 좀 더 js 스러운 코드가 될 것 같아요.

Johnie-Yeo commented on web/src/components/BottomMenu/index.js:
```diff
@@ -0,0 +1,29 @@
+                      e.preventDefault();
+                      this.props.onChangePage(id);
+                  }.bind(this, data[i].id)}
+                ><img className="Image" src={require(`../../images/${data[i].img}`)} /></button>)
```
코드가 돌아가는데는 문제가 없지만 동적으로 데이터를 불러오는 방식은 통용되지는 않는 방법이에요.
require의 경우는 nodejs가 사용하는 commonjs스펙이기도 하고요.
또 여기 list에 들어가는 하나의  버튼을 별개의 컴포넌트로 분리한다면 좀 더 좋아 보일 것 같다는 생각도 드네요.

Johnie-Yeo commented on web/src/components/Header/index.js:
```diff
@@ -0,0 +1,17 @@
+    render() {
+      return (
+        <header>
+          <h1><a href="/" onClick={function(e){
```
이벤트 함수를 별도로 분리한다면 더 보기 좋을것 같아요

Johnie-Yeo commented on web/src/components/Header/index.css:
```diff
@@ -0,0 +1,14 @@
+@import url('https://fonts.googleapis.com/css2?family=Jua&display=swap');
```
컴포넌트 내부의 css에는 하나의 컴포넌트에 종속적인 내용이 들어가면 더 좋을 것 같아요.
폰트의 경우 어플리케이션 전역에 쓰이니 App.css에 쓰이는게 더 좋지 않을까요?

Johnie-Yeo reviewed (changes requested):
고생 많이 하셨어요!!!
약간의 코멘트를 덧 붙이자면
1. 커밋은 작업의 단위입니다. 하나의 작업 = 1커밋 이 되도록 신경써본다면 좋을 것 같아요
2. Pull Request의 제목은 이 커밋들을 포괄하는 내용으로 적어주시면 더 좋을 것 같아요.
3. 진행하시는 내용에 대해 이슈 업데이트 및 zenhub 관리도 해주시면 보기 좋을 것 같아요.
처음 하시느라 엄청 힘드셨을 것 같아요!
너무 수고하셨고 조금만 더 보완해 보시면 좋을 것 같습니다! ㅎㅎ

rina214 committed:
feat: create homepage

홈 페이지의 검색창, 인기 꿀팁, 최근 변경 컴포넌트를 생성하였습니다.
warning이 아주 많습니다,,, :(

rina214 committed:
refactor: homepage refactoring

검색창이 다른 페이지에서도 쓰이기 때문에 component로 만들었습니다.

rina214 committed:
refactor: use router at bottom navigation

하단 내비게이션에서 버튼을 누르면 router를 통해 link가 바뀌도록 하였습니다.

rina214 committed:
fix: change homepage layout

와이어 프레임과 비슷하게 홈 화면을 구성하였습니다.
아직은 화면만 그린 상태입니다.

Johnie-Yeo commented on web/src/components/HashTag/index.jsx:
```diff
@@ -0,0 +1,8 @@
+import React from "react";
+import "./style.css";
+
+const HashTag = (props) => {
```
여기서도 {tag} 이런식으로 destructuring 해서 받을 수 있어요.

Johnie-Yeo commented on web/src/components/HomePage/index.jsx:
```diff
@@ -0,0 +1,67 @@
+import CategoryCollection from "../CategoryCollection";
+
+function HomePage() {
+  const data = {
```
이런 가짜 데이터 혹은 정적 데이터를 모으는 파일을 별도로 만드는 것도 좋은 방법입니다.
코드에서는 내부 로직만 보여줄 수 있도록 하는거죠.

Johnie-Yeo commented on web/src/components/HomePage/index.jsx:
```diff
@@ -0,0 +1,67 @@
+      <SearchBar />
+      <CategoryCollection />
+      <hr className="Line" />
+      <GoodTip title={data.hotPost.name} posts={data.hotPost.posts}></GoodTip>
```
GoodTip이라는 컴포넌트가 인기게시물과 최신 게시물을 모두 보여주는 컴포넌트네요.
컴포넌트 이름을 의미에 맞는 것으로 바꿔보는 것은 어떨까요?

Johnie-Yeo commented on web/src/components/Layout/index.jsx:
```diff
@@ -0,0 +1,45 @@
+
+const Layout = ({ children }) => {
+  const data = {
+    mode: 1,
```
이런 정적 데이터도 분리해 주세요.

Johnie-Yeo reviewed (approved):
코드가 이전에 비해 한결 깔끔해졌어요!
약간의 수정사항만 반영하시면 될 것 같네요

rina214 committed:
fix: fix the (import image) bug

public 폴더 내부에 있던 image를 import하려고 시도한 코드를 삭제했습니다.
이전 코드에서 저번엔 왜 돌아갔었는지 모르겠네요 ㅠㅠ

miue22 commented on web/src/App.js:
```diff
@@ -1,25 +1,34 @@
+    <Router className="App">
+      <Switch>
+        <Layout>
+          <Route exact path="/">
+            <HomePage />
+          </Route>
+          <Route exact path="/menu">
+            <MenuPage />
+          </Route>
+          <Route exact path="/writing">
+            <WritingPage />
+          </Route>
+          <Route exact path="/mypage">
+            <MyPage />
+          </Route>
```
Route로 렌더링하는게 단일 컴포넌트라면 아래와 같이 사용하는게 더 가독성이 좋을 것 같아요 ㅎ

```jsx
<Route exact path="/" component={HomePage} />
<Route exact path="/menu" component={MenuPage }>
```

miue22 commented on web/src/components/CategoryCollection/index.jsx:
```diff
@@ -0,0 +1,57 @@
+  return (
+    <div>
+      <h3>카테고리 모아보기</h3>
+      <div className="container">{categoryCollection}</div>
```
`categoryCollection`의 값이 JSX 뿐이라면 일반적인 함수나 변수보다는 컴포넌트라고 보아야 할 것 같은데, 네이밍을 `CategoryCollection`으로 변경하는게 좋겠네요~

miue22 commented on web/src/components/HomePage/index.jsx:
```diff
@@ -0,0 +1,67 @@
+      <hr className="Line" />
+      <GoodTip title={data.hotPost.name} posts={data.hotPost.posts}></GoodTip>
+      <hr className="Line" />
+      <GoodTip
```
태그 사이에 값이 아무것도 없다면, 아래와 같이 바로 닫아버릴 수 있어요 :)

```jsx
<GoodTip 
    title={data.recentPost.name} 
    posts={data.recentPost.posts}
/>
```

rina214 committed:
refactor: homePage and inCategory component

rina214 committed:
feat: connect with backend

재환님의 백엔드랑 연결하였습니다.
